### PR TITLE
[GCE] Update hostname to be unique

### LINKED ISSUE #1736: [GCE] Non unique hostname
The agent currently uses the instance name as the hostname of GCE instances.
If one has several projects with instances having the same name, this hostname isn't unique, thus the tags and metrics not matching the expected values.
A unique hostname would be either what's called `internal-hostname` or `name-project`

### PR DESCRIPTION
**Pull request by tmichelet**
fixes #1736 
- update the hostname to `host_metadata['instance']['hostname']` so it's unique
- drop some tags that we don't want to collect
- add `"instance_name.project_id"` as a host alias

### CODE REVIEW

File: datadog.conf.example [hunk 1]
```diff
@@ -43,6 +43,9 @@ api_key:
 # for more information
 # collect_instance_metadata: yes
 
+# use unique hostname for GCE hosts. see #1736
```
Comment (tmichelet) [inline datadog.conf.example]: let's put PR link

Comment (remh) [inline datadog.conf.example]: Ping me when this is done

Comment (tmichelet) [inline datadog.conf.example]: first pass is done, see https://github.com/DataDog/dd-agent/wiki/Hostname-on-Google-Cloud-Engine

**Comment by tmichelet** [comment]
@remh link updated, good to merge?

**Pull request merged**